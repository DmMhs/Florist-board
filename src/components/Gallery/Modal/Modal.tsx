import React, { Component } from 'react';

import './Modal.less';
import LeftArrow from '../../Slider/LeftArrow/LeftArrow';
import RightArrow from '../../Slider/RightArrow/RightArrow';

interface ModalProps {
  images: JSX.Element[];
  initial: number;
  prevClickedHandler:
    | ((event: React.MouseEvent<HTMLDivElement, MouseEvent>) => void)
    | undefined;
  nextClickedHandler:
    | ((event: React.MouseEvent<HTMLDivElement, MouseEvent>) => void)
    | undefined;
  closeModal:
    | ((event: React.MouseEvent<HTMLDivElement, MouseEvent>) => void)
    | undefined;
}
interface ModalState {
  activeSlide: JSX.Element | null;
  current: number | null;
}

class Modal extends Component<ModalProps, ModalState> {
  static getDerivedStateFromProps(props: ModalProps, state: ModalState) {
    const { images, initial } = props;
    return {
      activeSlide: images[initial],
      current: initial
    };
  }
  constructor(props: ModalProps) {
    super(props);
    this.state = {
      activeSlide: null,
      current: null
    };
  }

  componentDidMount() {
    const { images, initial } = this.props;
    this.setState({
      activeSlide: images[initial]
    });
  }

  render() {
    return (
      <div className="Modal">
        <div className="modal-content">
          <i className="fas fa-times close" onClick={this.props.closeModal} />
          <div className="slide-wrapper">
            {this.state.activeSlide !== null ? this.state.activeSlide : null}
          </div>
          <LeftArrow
            goToPrevSlide={this.props.prevClickedHandler}
            show={true}
          />
          <RightArrow
            goToNextSlide={this.props.nextClickedHandler}
            show={true}
          />
          <span className="current-count">
            {this.state.current! + 1}/{this.props.images.length}
          </span>
        </div>
      </div>
    );
  }
}

export default Modal;
