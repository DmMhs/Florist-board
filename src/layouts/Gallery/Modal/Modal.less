.Modal {
  position: fixed;
  top: 0;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 100%;
  height: 100%;
  background-color: rgba(51, 51, 51, 0.5);
  z-index: 9;
  .modal-content {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    position: relative;
    width: 75%;
    min-height: 60%;
    border-radius: 7px;
    background-color: white;
    z-index: 4;
    .close {
      position: absolute;
      top: 20px;
      right: 25px;
      font-size: 1.3rem;
      color: #333333;
      z-index: 4;
      cursor: pointer;
      &:hover {
        color: tomato;
      }
    }
    .slide-wrapper {
      .image {
        max-width: 500px;
      }
    }
    .current-count {
      position: absolute;
      bottom: 20px;
      right: 25px;
    }
  }
}
@media screen and (max-width: 768px) {
  .Modal {
    .modal-content {
      width: 300px;
      height: calc(100% - 1rem);
      .slide-wrapper {
        .image {
          max-width: 290px;
        }
      }
    }
  }
}
