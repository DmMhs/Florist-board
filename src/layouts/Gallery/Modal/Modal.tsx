import React, { Component } from 'react';

import './Modal.less';
import LeftArrow from '../../../components/Slider/LeftArrow/LeftArrow';
import RightArrow from '../../../components/Slider/RightArrow/RightArrow';

interface ModalProps {
  images: JSX.Element[];
  initial: number;
  prevClickedHandler:
    | ((event: React.MouseEvent<HTMLDivElement, MouseEvent>) => void)
    | undefined;
  nextClickedHandler:
    | ((event: React.MouseEvent<HTMLDivElement, MouseEvent>) => void)
    | undefined;
  closeModal:
    | ((event: React.MouseEvent<HTMLDivElement, MouseEvent>) => void)
    | undefined;
}
interface ModalState {
  activeSlide: JSX.Element | null;
  current: number | null;
}

class Modal extends Component<ModalProps, ModalState> {
  public static getDerivedStateFromProps(props: ModalProps, state: ModalState) {
    const { images, initial } = props;
    return {
      activeSlide: images[initial],
      current: initial
    };
  }
  constructor(props: ModalProps) {
    super(props);
    this.state = {
      activeSlide: null,
      current: null
    };
  }

  public componentDidMount() {
    const { images, initial } = this.props;
    this.setState({
      activeSlide: images[initial]
    });
  }

  public render() {
    const {
      closeModal,
      prevClickedHandler,
      nextClickedHandler,
      images
    } = this.props;

    const { activeSlide, current } = this.state;

    return (
      <div className="Modal">
        <div className="modal-content">
          <i className="fas fa-times close" onClick={closeModal} />
          <div className="slide-wrapper">
            {activeSlide !== null ? activeSlide : null}
          </div>
          <LeftArrow goToPrevSlide={prevClickedHandler} show={true} />
          <RightArrow goToNextSlide={nextClickedHandler} show={true} />
          <span className="current-count">
            {current! + 1}/{images.length}
          </span>
        </div>
      </div>
    );
  }
}

export default Modal;
