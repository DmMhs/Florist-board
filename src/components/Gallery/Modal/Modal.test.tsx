import React from 'react';
import { shallow, mount } from 'enzyme';

import Modal from './Modal';

it('Modal component matches a snapshot', () => {
  const wrapper = shallow(
    <Modal
      images={[]}
      initial={1}
      prevClickedHandler={() => {}}
      nextClickedHandler={() => {}}
      closeModal={() => {}}
    />
  );
  expect(wrapper).toMatchSnapshot();
});

it('Active slide is displayed', () => {
  const wrapper = shallow(
    <Modal
      images={[<img className="image" />, <img className="image" />]}
      initial={1}
      prevClickedHandler={() => {}}
      nextClickedHandler={() => {}}
      closeModal={() => {}}
    />
  );

  expect(wrapper.find('.image').exists()).toBeTruthy();

  const wrapper2 = shallow(
    <Modal
      images={[]}
      initial={1}
      prevClickedHandler={() => {}}
      nextClickedHandler={() => {}}
      closeModal={() => {}}
    />
  );

  expect(wrapper2.find('.image').exists()).toBeFalsy();
});
