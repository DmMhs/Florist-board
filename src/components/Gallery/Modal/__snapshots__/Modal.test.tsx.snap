// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component matches a snapshot 1`] = `
<div
  className="Modal"
>
  <div
    className="modal-content"
  >
    <i
      className="fas fa-times close"
      onClick={[Function]}
    />
    <div
      className="slide-wrapper"
    />
    <LeftArrow
      goToPrevSlide={[Function]}
      show={true}
    />
    <RightArrow
      goToNextSlide={[Function]}
      show={true}
    />
    <span
      className="current-count"
    >
      2
      /
      0
    </span>
  </div>
</div>
`;
